### ISSUE TITLE
Cannot connect to sharepoint server after the new update

### ISSUE THREAD

**Issue by AlexanderBodnar1** [vgrem/Office365-REST-Python-Client#596]
My problem is that i cannot Log in to the sharepoint server.

Here is the code :
![image](https://user-images.githubusercontent.com/116795167/201024660-68081bbd-60c1-40ea-af3c-7d919d989c38.png)

Here is the error:
![image](https://user-images.githubusercontent.com/116795167/201024810-618ec063-a38e-4772-9e3b-8fac4e153d68.png)
![image](https://user-images.githubusercontent.com/116795167/201024873-8573bd56-afc6-445b-b76e-98dd5cd8bd2a.png)

Basically the first error is this: 
![image](https://user-images.githubusercontent.com/116795167/201024967-f1c28135-a368-4e79-a7a5-f3be76a68eab.png)

The urllib3 package exists. I updated it to the newest version aswell. Whats the solution for this?

Thanks for your respond,
Alex

**Comment by AlexanderBodnar1** [comment]
Can somebody help me? I cannot manage to fix it

**Comment by florinmihut-intel** [comment]
I see in your screenshots that getaddrinfo failed. This typically means that you can't solve the host you are trying to connect (login.microsoftonline.com). Are you behind a proxy? Are you connected to the internet on the computer where you are executing this? If you ping login.microsoftonline.com on that computer do you see the host being resolved?

**Comment by florinmihut-intel** [comment]
Since getaddrinfo returns an error, I don't think this is related to the python module. Provide a bit more info about what are you doing. Are you running this on Win or Linux env? Are the environment variables set for the http/https proxy set? Can you ping login.microsoftonline.com? (at least do you see an IP when pinging?)

**Comment by AlexanderBodnar1** [comment]
Im running this on Win with a virtualenv created. I already figured out that it's not the problem with the python module. Urllib3 manages to create dependencies dynamically so that's why the Editor shows the error, because it cannot see them.

I have an app which downloads certain files from the sharepoint server and uploads them back. Before that i need to log-in to the sharepoint site. I'm doing it with user credentials.

When pinging in cmd it gives me this result:
![image](https://user-images.githubusercontent.com/116795167/203560444-bbbdc004-1d77-4ba8-928e-00457cbde42b.png)

**Comment by florinmihut-intel** [comment]
The problem is that there is no route from your computer to login.microsoftonline.com. It can't resolve the hostname. Maybe your network configuration has a problem?! If ping doesn't resolve that name, you could try nslookup login.microsoftonline.com? Don't put the screenshot here because that might tell something about your network, but generally, you should see the server that is resolving the hostnames and it should provide a list of IPs and a list of aliases for the hostname you are searching. Contact your admin, you have a configuration problem while on VPN in my opinion. You should try the nslookup command with VPN and without to see the difference. 

@vgrem This is not a problem with the python module and should be closed.
